numba/tests/test_fancy_indexing.py

```diff
@@ -255,6 +255,41 @@ def check(arr, ind):
 
         #exceptions leak refs
         self.disable_leak_check()
+    
+    def test_new_axis(self):
+        def np_new_axis_getitem(a, idx):
+            return a[idx]
+
+        def np_new_axis_setitem(a, idx, item):
+            a[idx] = item
+            return a
+
+        a = np.arange(4*5*6*7).reshape((4, 5, 6, 7))
+        idx_cases = [
+            (np.newaxis, slice(None)),
```

Comment:
As above, perhaps add some cases where `np.newaxis` isn't the first item?

Reply:
Addressed in 89ba549